shared/test/test_files.rb

```diff
@@ -548,6 +548,142 @@ def test_files_copy_all
     delete_channel(dest_channel_id)
   end
 
+  def test_temporary_public_url
+    thumbnail_filename = '.metadata/thumbnail.png'
+    thumbnail_body = 'stub-thumbnail-contents'
+
+    @api.put_object(thumbnail_filename, thumbnail_body)
+    assert successful?
+
+    temp_url = FileBucket.new.make_temporary_public_url(@channel_id, thumbnail_filename)
+
+    # Links to the right file
+    assert_match /#{'cdo-v3-files.s3.amazonaws.com/files_test/1/1/.metadata/thumbnail.png'}/, temp_url
+
+    # Has a 5-minute timeout by default
+    assert_match /#{'X-Amz-Expires=300'}/, temp_url
+
+    @api.delete_object(thumbnail_filename)
+    assert successful?
+  end
+
+  def test_temporary_public_url_with_custom_timeout
+    thumbnail_filename = '.metadata/thumbnail.png'
+    thumbnail_body = 'stub-thumbnail-contents'
+
+    @api.put_object(thumbnail_filename, thumbnail_body)
+    assert successful?
+
+    temp_url = FileBucket.new.make_temporary_public_url(@channel_id, thumbnail_filename, 1.hour)
+
+    # Links to the right file
+    assert_match /#{'cdo-v3-files.s3.amazonaws.com/files_test/1/1/.metadata/thumbnail.png'}/, temp_url
+
+    # Has a 5-minute timeout by default
+    assert_match /#{'X-Amz-Expires=3600'}/, temp_url
+
+    @api.delete_object(thumbnail_filename)
+    assert successful?
+  end
+
+  def test_adult_thumbnail
+    ImageModeration.stubs(:rate_image).returns :adult
+    thumbnail_filename = '.metadata/thumbnail.png'
+    thumbnail_body = 'stub-adult-thumbnail'
+
+    # Intentional: Not yet checking/blocking adult content on upload.
+    @api.put_object(thumbnail_filename, thumbnail_body)
+    assert successful?
+
+    get "/v3/files-public/#{@channel_id}/#{thumbnail_filename}"
+
+    # Responds with a 404, like we do for flagged content
+    assert not_found?
+
+    # Includes content rating metadata in the response that the client can read
+    assert_equal 'adult', last_response['x-cdo-content-rating']
+
+    # Response is cached for an hour
+    assert_equal 'public, max-age=3600, s-maxage=1800', last_response['Cache-Control']
+
+    # Flags the project as abusive.
+    get "/v3/channels/#{@channel_id}/abuse"
+    assert successful?
+    assert_operator 10, :<=, JSON.parse(last_response.body)['abuse_score']
+
+    # Flags the asset as abusive
```

Comment:
probably better to say that it flags the thumbnail as abusive, here and below. I got confused here at first because we said earlier we aren't flagging assets yet.